AzureDevOps.Authentication/Src/Authentication.cs

```diff
@@ -131,31 +131,11 @@ public override async Task<bool> DeleteCredentials(TargetUri targetUri)
             if (targetUri is null)
                 throw new ArgumentNullException(nameof(targetUri));
 
-            Credential credentials = await PersonalAccessTokenStore.ReadCredentials(targetUri);
+            await PersonalAccessTokenStore.DeleteCredentials(targetUri);
 
-            // Attempt to validate the credentials, if they're truly invalid delete them.
```

Comment:
I am not sure you can safely just do this. Remember, if a user has more than one credential helper registered git-credential will invoke them in order. However, git-credential will only accept credentials from the first helper which claims to have them. If those credentials fails (for whatever reason) git-credential will then call every registered helper with an erasure request.

For many users, this is a disaster. The entire point of the `ValidateCredentials` call was avoid deleting good credentials.

Previously, I'd mentioned that there could be a disconnect between how GCM validates credentials (detecting that they're good) and what privileges those credentials provide (good token, but it does not grant access to a given repository - for example).

I was more hoping that you could modernize the `ValidateCredentials` method to utilize the more accurate `targetUri.ActualUri` property to validate the credentials with; and no so much give up on protecting users.